core/state.go

```diff
@@ -492,6 +520,24 @@ func (s *State) Revert(blockNumber uint64, update *StateUpdate) error {
 		}
 	}
 
+	// purge noClassContracts
+	//
+	// As noClassContracts are not in StateDiff.DeployedContracts we can only purge them if their storage no longer exists.
+	// Updating contracts with reverse diff will eventually lead to the deletion of noClassContract's storage key from db. Thus,
+	// we can use the lack of key's existence as reason for purging noClassContracts.
+
+	for addr := range noClassContracts {
+		err := s.txn.Get(db.ContractStorage.Key(addr.Marshal()), func(val []byte) error {
```

Comment:
This leaks `contract` package implementation. Can we use `if contract.Root() == felt.Zero` instead?

Reply:
Done.

Although I don't think its `leakage` as they are in the same package.